lib/linux_admin/hosts.rb

```diff
@@ -36,6 +38,20 @@ def update_entry(address, hostname, comment = nil)
       end
     end
 
+    def hostname=(name)
+      if cmd?("hostnamectl")
+        run!(cmd('hostnamectl'), :params => ['set-hostname', name])
+      else
+        File.write("/etc/hostname", name)
+        run!(cmd('hostname'), :params => {:file => "/etc/hostname"})
+      end
+    end
+
+    def hostname
+      result = run(cmd("hostname"))
```

Comment:
We already require `activesupport/core_ext` so you could simplify this if there is nothing on output when unsuccessful.
```ruby
run(cmd("hostname")).output.presence
```